doc/source/docs/user_guide/fiona_to_pyogrio.rst

```diff
@@ -0,0 +1,71 @@
+Migration from the Fiona to the Pyogrio IO engine
+=================================================
+
+Since version 0.11, GeoPandas supports two IO engines to read and write files:
+`Fiona <https://fiona.readthedocs.io>`__ and
+`Pyogrio <https://pyogrio.readthedocs.io>`__.
+
+For Geopandas versions <1.0, GeoPandas defaults to use Fiona. It is possible to use
+Pyogrio using the ``engine="pyogrio"`` parameter in :func:`geopandas.read_file` and
+:func:`geopandas.GeoDataFrame.to_file`. You can also change the default engine globally
+with:
+
+.. code-block:: python
+
+    geopandas.options.io_engine = "pyogrio"
+
+Starting from GeoPandas 1.0, the global default will change from Fiona to Pyogrio. The
+main reason for this change is performance. Pyogrio is optimized for the use case
+relevant for GeoPandas: reading and writing in bulk. Because of this, in many cases
+speedups >5-20x can be observed.
+
+This guide outlines the (known) functional differences between both, so you can account
+for them when switching to Pyogrio.
+
+
+Write an attribute table to a file
+----------------------------------
+
+Using the Fiona IO engine, it was possible to write an attribute table (a table without
+geometry column) to a file using the ``schema`` parameter to specify that the "geometry"
+column should be ignored.
+
+With Pyogrio you can write an attribute table by calling
+:func:`geopandas.GeoDataFrame.to_file` on a GeoDataFrame without geometry column:
+
+.. code-block:: python
+
+    >>> import geopandas as gpd
+    >>> 
+    >>> gdf = gpd.GeoDataFrame({"data_column": [1, 2, 3]})
+    >>> gdf.to_file("test_attribute_table.gpkg", engine="pyogrio")
```

Comment:
I don't think we can recommend this pattern - in general a GeoDataFrame without a geometry column will be a DataFrame instead. The constructor is a special case. But any slicing operation e.g. `gdf[:]` will downcast to a dataframe. I think either we need to suggest `pyogrio.write_dataframe`, add our own toplevel wrapper for this, or monkey patch to_file onto `pd.DataFrame` like we do for `set_geometry` (though this seems like it will lead to confusion, so I'd rather avoid it)

Reply:
> I think either we need to suggest pyogrio.write_dataframe

+1 on that.

Reply:
@jorisvandenbossche to be sure... can you have a quick look at this? In short: a GeoDataFrame without geometry column is not to be recommended?

Reply:
No, it is shall not be used. We point to pyogrio's read_dataframe in docs when talking about non-spatial layers already, so we can point to `write_dataframe` here when trying to write them.